includes/models/newspaper.inc

```diff
@@ -1,42 +1,99 @@
   module_load_include('inc', 'islandora_paged_content', 'includes/utilities');
   module_load_include('inc', 'islandora_newspaper', 'includes/utilities');
+  module_load_include('inc', 'islandora', 'includes/datastream');
+  
+  if (is_null($selected_collection)) {
+    $manifest = islandora_iiif_manifest_get_newspaper_manifest($object);
+  }
+  else {
+    $manifest = islandora_iiif_manifest_get_newspaper_collection($object, $selected_collection);
+  }
+  
+  return $manifest;
+}
 
-  // Manifest array.
-  $manifest = array();
-
+/**
+ * Create the newspaper manifest with collections and its member within
+ * @param $object
+ * @return mixed
+ */
+function islandora_iiif_manifest_get_newspaper_manifest($object) {
+  // Collections array
+  $manifest['collections'] = array();
   // Get issues.
   $issues = islandora_newspaper_get_issues($object);
-
-  $collection_members_members = array();
-
-  foreach ($issues as $issue) {
-
-    // Load issue.
-    $issue_object = islandora_object_load($issue['pid']);
-
-    $collection_item = islandora_iiif_manifests_get_object_collection($issue_object);
-
-    $collection_members_members[] = $collection_item['object'];
-
+  // Get issues.
+  $group_issues = islandora_newspaper_group_issues($issues);
+  // Collections array
+  $manifest['collections'] = array();
+  foreach ($group_issues as $collection => $member) {
+    
+    $newspaper_member = islandora_iiif_manifests_get_newspaper_member_collection($member);
+    if (!empty($newspaper_member)) {
+      $manifest_uri = urldecode(url(variable_get('islandora_iiif_manifests_domain_uri', $GLOBALS['base_url']) . "/iiif_manifest/{$object->id}/collection/{$collection}", array(
+        'absolute' => TRUE,
+        'language' => (object) array('language' => FALSE),
+      )));
+      if (is_numeric($collection)) {
+        $label = strval($collection);
+      }
+      else {
+        $label = $collection;
+      }
+      $newspaper_collection = array(
+        "@id" => $manifest_uri,
+        "@type" => 'sc:Collection',
+        "label" => $label,
+      );
+      $newspaper_collection['members'] = $newspaper_member;
+      array_push($manifest['collections'], $newspaper_collection);
+    }
   }
+  
+  return $manifest;
+}
 
-  if (!empty($collection_members_members)) {
-    $manifest['members'] = $collection_members_members;
-  }
+/**
+ * Create the newspaper manifest collection
+ * @param $object
+ * @param $selected_collection
+ * @return array
+ */
+function islandora_iiif_manifest_get_newspaper_collection($object, $selected_collection) {
+  
+  // Members array
+  $manifest['members'] = array();
+  
+  // Get IIIF Manifest
+  $manifest_ds = json_decode($object[ISLANDORA_IIIF_MANIFESTS_DATASTREAM_ID]->content);
+  
+  // Retrieve the collection member based on the collection label
+  foreach ($manifest_ds->collections as $col) {
+    
+    $collection_id = str_replace("'", "", $col->label);
```

Comment:
Is the $col->label always the issue year (I think you are trying to accomplish this)? I do not think you can count on that...

Reply:
@lucasvanschaik, no. The reason why I need the label is to match is with the selected part label. I do a "str_replace" to remove the single quotes.